packages/block-library/src/heading/edit.js

```diff
@@ -61,7 +69,17 @@ function HeadingEdit( {
 				value={ content }
 				onChange={ ( value ) => setAttributes( { content: value } ) }
 				onMerge={ mergeBlocks }
-				onSplit={ ( value ) => {
+				onSplit={ ( value, keepId ) => {
+					if ( keepId ) {
```

Comment:
Do we really want to have the "keepId" argument? I mean why not just replace the `createBlock( 'core/heading'` call a few lines down here?

Reply:
Yes, we still need the `keepId` since it determines if we should use the original block or create a new one. W/o that we wouldn't know if we should create a new block or make changes in the "original" one. I am wondering if I could use `getBlock` in the `rich-text` component and use `onSplit` only for creating new blocks but I'm not sure if it's a good way because we would lose the "flexibility" in that mechanism (currently it could work differently for heading, paragraph, list etc).

Reply:
Adding `keepId` is a horrible API. We should go for one or the other for all cases. Any alternatives we can explore here?

Reply:
>Adding keepId is a horrible API. We should go for one or the other for all cases.

Can you elaborate a bit? :)

IMHO it is still better than removing the original block and replacing it. Even from the performance perspective update is faster than unmount + mount (we use the `clientId` as a key in our block-list). 

>Any alternatives we can explore here?

As I mentioned I was wondering if it could be done inside `rich-text/index.js`. We could get block and change attributes inside instead of calling `onSplit` for the original block but i haven't investigated it.